src/vaadin-grid-data-provider-mixin.html

```diff
@@ -231,10 +231,14 @@
     }
 
     _isExpanded(item) {
-      return this.expandedItems && this._getItemIndexInArray(item, this.expandedItems) > -1;
+      return !!this.__expandedKeys.get(this.getItemId(item));
     }
 
     _expandedItemsChanged(e) {
+      this.__expandedKeys = new Map();
+      this.expandedItems.forEach(item => {
+        this.__expandedKeys.set(this.getItemId(item), true);
+      });
```

Comment:
Wondering if a `Set` would work better (the values in the map are always `true` for each item)

Reply:
Done